src/js/msp.js

```diff
@@ -147,25 +234,33 @@ var MSP = {
     clearListeners: function() {
         this.listeners = [];  
     },
-    send_message: function (code, data, callback_sent, callback_msp, callback_onerror) {
-        if (code === undefined) {
-            debugger;
+    crc8_dvb_s2: function(crc, ch) {
+        crc ^= ch;
+        for (let ii = 0; ii < 8; ii++) {
+            if (crc & 0x80) {
+                crc = ((crc << 1) & 0xFF) ^ 0xD5;
+            } else {
+                crc = (crc << 1) & 0xFF;
+            }
+        }
+        return crc;
+    },
+    crc8_dvb_s2_data: function(data, start, end) {
+        let crc = 0;
+        for (let ii = start; ii < end; ii++) {
+            crc = this.crc8_dvb_s2(crc, data[ii]);
         }
-        var bufferOut,
-            bufView;
-
-         if (!callback_onerror) {
-             var callbackOnError = false;
-         } else {
-             var callbackOnError = true;
-         }
+        return crc;
+    },
+    encode_message_v1: function(code, data) {
+        let bufferOut;
         // always reserve 6 bytes for protocol overhead !
         if (data) {
             var size = data.length + 6,
                 checksum = 0;
 
             bufferOut = new ArrayBuffer(size);
-            bufView = new Uint8Array(bufferOut);
+            let bufView = new Uint8Array(bufferOut);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['bufView' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/emuflight/EmuConfigurator/pullRequest?prid=5635143)